embassy-executor/build.rs

```diff
@@ -3,6 +3,9 @@ use std::fmt::Write;
 use std::path::PathBuf;
 use std::{env, fs};
 
+#[path = "../build_common.rs"]
```

Comment:
unfortunately this breaks `cargo publish`. all files from a crate have to be present under the crate root: 

```
[dirbaio@mars embassy-executor]$ cargo publish --dry-run
    Updating crates.io index
   Packaging embassy-executor v0.5.0 (/home/dirbaio/embassy/embassy/embassy-executor)
   Verifying embassy-executor v0.5.0 (/home/dirbaio/embassy/embassy/embassy-executor)
    Updating crates.io index
   Compiling proc-macro2 v1.0.84
   Compiling unicode-ident v1.0.12
   Compiling ident_case v1.0.1
   Compiling fnv v1.0.7
   Compiling strsim v0.11.1
   Compiling embassy-executor v0.5.0 (/home/dirbaio/embassy/embassy/embassy-executor/target/package/embassy-executor-0.5.0)
   Compiling litrs v0.4.1
   Compiling critical-section v1.1.2
error: couldn't read ../build_common.rs: No such file or directory (os error 2)
 --> build.rs:7:1
  |
7 | mod common;
  | ^^^^^^^^^^^

error: could not compile `embassy-executor` (build script) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: failed to verify package tarball
```

unfortunately I can't see a good solution... i'd just copypaste the stuff. We could make a shared crate but then we have to publish it, version it, etc. I'm not sure if it's worth it.

Reply:
Oh, good catch! What about adding a symlink `embassy-executor/build_common.rs` -> `../build_common.rs` and refering to `./build_common.rs` (the symlink) from `build.rs`? That seems to work with `cargo publish` and doesn't require copy-pasting the same file to multiple directories.

Reply:
I pushed the symlink solution, but I will change it to copy-paste if there is some issue with symlinks that I haven't foreseen.

Reply:
do symlinks work if you clone the repo on Windows?

Reply:
Ok, it's copy-pasta, then :)